core/app/models/spree/address.rb

```diff
@@ -191,7 +191,8 @@ def state_validate
       # ensure state_name belongs to country without states, or that it matches a predefined state name/abbr
       if state_name.present?
         if country.states.present?
-          states = country.states.find_all_by_name_or_abbr(state_name)
+          state_lookup = state_name.downcase.strip
+          states = country.states.where('lower(name) = ? or lower(abbr) = ?', state_lookup, state_lookup);
```

Comment:
This should use AREL `matches` instead of `LOWER`.


```Ruby
states = country.states.where(
  Spree::State.arel_table[:name].matches(state_name).or(
    Spree::State.arel_table[:name].matches(state_name)
  )
)
```

That will be properly database agnostic and potentially allow usage of indexes.